src/screens/Dashboard/DashboardScreen.tsx

```diff
@@ -59,33 +59,27 @@ class DashboardScreen extends Component<Props, State> {
       .catch(() => {
         this.props.navigation.navigate(Route.CreatePin);
       });
-    this.props.loadWallets();
+    await this.props.loadWallets();
   }
 
   refreshTransactions = async () => {
     this.setState({ isFetching: true });
-    const { lastSnappedTo } = this.state;
-    const { wallets } = this.props;
-    if (isAllWallets(wallets[lastSnappedTo])) {
-      await this.props.loadWallets();
-    } else {
-      await this.props.loadTransactions(wallets[lastSnappedTo].secret);
-    }
+    await this.props.loadWallets();
+
     this.setState({ isFetching: false });
   };
 
   chooseItemFromModal = (index: number) => {
     this.setState({ lastSnappedTo: index });
   };
 
-  onSnapToItem = async (index: number) => {
-    this.refreshWallet(index);
+  onSnapToItem = async () => {
+    await this.refreshWallet();
   };
 
-  refreshWallet = (index: number) => {
+  refreshWallet = async () => {
```

Comment:
you could also return promise `return this.props.loadWallets()` and then you would not need async/await